mteb/cache.py

```diff
@@ -275,6 +279,160 @@ def download_from_remote(
 
         return results_directory
 
+    def download_cached_results_from_branch(
+        self,
+        branch: str = "cached-data",
+        filename: str = "__cached_results.json.gz",
+        output_path: Path | None = None,
+        remote: str = "https://github.com/embeddings-benchmark/results",
```

Comment:
do we want this to work on any other repo (it requires the exact setup that we have)

Reply:
It might be better if I unify `remote: str = "https://github.com/embeddings-benchmark/results"` from `download_from_remote` ?

Reply:
> could work with maeb-results, for instance, of the human-benchmark-results.

but HUME is merged and MAEB will probably also be added here - it would still require a us to setup the CI for those repos.


> It might be better if I unify

not entirely sure what you mean here

Reply:
I'm referring to the `download_from_remote()` method also referencing the same remote URL. If we don't care to offer the option to read from a different results repo (think of users who hosts this in their own org or fork), then we can set a single `remote_url` param and share between the 2 methods.